query/src/main/scala/filodb/query/exec/MultiSchemaPartitionsExec.scala

```diff
@@ -99,6 +99,12 @@ final case class MultiSchemaPartitionsExec(queryContext: QueryContext,
   def doExecute(source: ChunkSource,
                 querySession: QuerySession)
                (implicit sched: Scheduler): ExecResult = {
+
+    // timeout can occur here if there is a build up in actor mailbox queue and delayed delivery
+    val queryTimeElapsed = System.currentTimeMillis() - queryContext.submitTime
+    if (queryTimeElapsed >= queryContext.queryTimeoutMillis)
+      throw QueryTimeoutException(queryTimeElapsed, this.getClass.getName)
```

Comment:
I don't think MultiSchemaPartExec is the best place to do that.  Not all queries have MSPE in its query plan.  Instead, I want to run this in the QueryActor when it first gets the ExecPlan.  If it is done here, then futures have already dispatched in the thread pool and some resources have already been taken up.

Reply:
Good point. Addressed.